flask_appbuilder/tests/test_api.py

```diff
@@ -3100,3 +3100,350 @@ def test_before_request(self):
         uri = "api/v1/model1beforerequest/update"
         rv = self.auth_client_put(client, token, uri, {})
         self.assertEqual(rv.status_code, 404)
+
+
+class UserAPITestCase(FABTestCase):
+    def setUp(self):
+        from flask import Flask
+        from flask_appbuilder import AppBuilder
+
+        self.app = Flask(__name__)
+        self.basedir = os.path.abspath(os.path.dirname(__file__))
+        self.app.config.from_object("flask_appbuilder.tests.config_api")
+        self.app.config["ENABLE_USER_CRUD_API"] = True
+        self.db = SQLA(self.app)
+        self.appbuilder = AppBuilder(self.app, self.db.session)
+
+        # TODO: this heinous hack is to avoid using stale db session leaking from
+        # RolePermissionAPITestCase
+        # don't know why all baseviews in Appbuilder are attached to stale session,
+        # causing error when adding a new user which reads roles from this session and
+        # datamodel uses stale session to add it.
```

Comment:
Don't really know why this happens.
If I run this test before `RolePermissionAPITestCase` it works fine, but if I run `RolePermissionAPITestCase` first then `datamodel` objects in the new `modelRestApi` classes already have session attached, so this line https://github.com/dpgaspar/Flask-AppBuilder/blob/8a18281002d55a50416cf98dad8d75c4c218dbb8/flask_appbuilder/base.py#L343 skips assigning current session to it.